src/pick.ts

```diff
@@ -26,17 +27,17 @@ export function pick<T extends object, K extends keyof T>(
  */
 export function pick<K extends PropertyKey>(
   names: ReadonlyArray<K>
-): <T extends Record<PropertyKey, any>>(object: T) => Pick<T, K>;
+): <T extends Record<K, unknown>>(object: T) => Unwrap<Pick<T, K>>;
 
 export function pick() {
   return purry(_pick, arguments);
 }
 
-function _pick(object: any, names: Array<string>) {
+function _pick(object: Record<string, unknown>, names: Array<string>) {
```

Comment:
shouldn't this be PropertyKey (it doesn't matter much, just for correctness)
```suggestion
function _pick(object: Readonly<Record<PropertyKey, unknown>>, names: ReadonlyArray<PropertyKey>) {
```